src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -778,7 +787,24 @@ trait SearchConverterService {
       }
     }
 
-    private def getTaxonomyResourceAndTopicsForId(id: Long, bundle: Bundle, taxonomyType: String) = {
+    private[service] def getGrepContexts(grepCodes: Seq[String], bundle: GrepBundle): List[SearchableGrepContext] = {
+      val grepContext = bundle.kjerneelementer ++ bundle.kompetansemaal ++ bundle.tverrfagligeTemaer
+
+      grepCodes
+        .map(
+          grepCode =>
+            SearchableGrepContext(
+              grepCode,
+              grepContext
+                .find(grepElement => grepElement.kode == grepCode) match {
+                case Some(element) => element.tittel.find(title => title.spraak == "default").map(title => title.verdi)
+                case None          => None
+              }
```

Comment:
Her hadde det vært mer elegant å (flat)mappe over optionen enn å matche på den:
```suggestion
              grepContext
                .find(grepElement => grepElement.kode == grepCode).flatMap(element => 
                  element.tittel.find(title => title.spraak == "default").map(_.verdi)  
                )
```

Hadde kanskje vært ryddig å dra det ut i en funksjon eller variabel og :)

Reply:
er det fordi at man da implisitt håndterer `case None` eller ?

Reply:
I guess du kan kalle det det.
```
opt.map(f)
```
er det samme som
```
opt match {
  case Some(x) => Some(f(x))
  case None    => None
}
```

scaladoc'en for standardbiblioteket er pretty good. Test å ctrl-klikk på `Option.map` i intellij :+1: